tests/unit/s2n_signature_algorithms_test.c

```diff
@@ -1396,6 +1373,106 @@ int main(int argc, char **argv)
         };
     };
 
+    EXPECT_SUCCESS(s2n_reset_tls13_in_test());
+
+    /* Self-Talk test: ECDSA signature scheme curves must only match certificate
```

Comment:
Idk I really struggled to write the test name / assertions. I meant "only" in that it didn't have to match the curve used for kex or the hash used for the PRF. But I can see how "only" implies that it CAN'T match those other things, which it could. They're just independent.